lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -153,36 +155,35 @@ function dosomething_northstar_update_user($user, $password = null) {
 }
 
 /**
- * Get user profile data from Northstar for the given Drupal ID.
+ * Get user profile data from Northstar by the specified ID.
  *
- * @param int $drupal_id Drupal user id.
- * @return object
+ * @param  string $id
+ * @param  string $type
+ * @return DoSomething\Northstar\Resources\NorthstarUser|null
  */
-function dosomething_northstar_get_northstar_user($drupal_id) {
-  $northstar_user_info = cache_get('northstar_user_info_' . $drupal_id, 'cache_dosomething_northstar');
+function dosomething_northstar_get_user($id, $type = 'drupal_id') {
+  $cache_bin = 'cache_dosomething_northstar';
+  $cache_key = 'northstar_user_'.$type.'='.$id;
+
+  $northstar_user = cache_get($cache_key, $cache_bin);
 
-  if ($northstar_user_info) {
-    $northstar_user_info = $northstar_user_info->data;
+  if ($northstar_user) {
+    return $northstar_user->data;
```

Comment:
Here's an example of how the record is stored in cache _after_ Drupal adds a few things to it.
```
stdClass Object
(
    [created] => 1475173802
    [expire] => 1475260201
    [data] => DoSomething\Northstar\Resources\NorthstarUser Object
        (
            [attributes:protected] => Array
                (
                    [id] => 57e161ffbffebc99278b4b7c
                    [_id] => 57e161ffbffebc99278b4b7c
                    [first_name] => Jen
                    [last_name] => a
                    [last_initial] => A
                    [photo] => 
                    [email] => jng@dosomething.org
                    [mobile] => 
                    [facebook_id] => 
                    [interests] => 
                    [birthdate] => 1986-12-31 00:00:00
                    [addr_street1] => 
                    [addr_street2] => 
                    [addr_city] => 
                    [addr_state] => 
                    [addr_zip] => 
                    [source] => phoenix
                    [mobilecommons_id] => 
                    [parse_installation_ids] => 
                    [mobilecommons_status] => 
                    [language] => en
                    [country] => 
                    [drupal_id] => 1707451
                    [role] => user
                    [updated_at] => 2016-09-20T16:21:19+0000
                    [created_at] => 2016-06-01T14:39:44+0000
                )

            [dates:protected] => Array
                (
                    [0] => created_at
                    [1] => updated_at
                )

        )

    [cid] => northstar_user_drupal_id=1707451
    [serialized] => 1
)
```